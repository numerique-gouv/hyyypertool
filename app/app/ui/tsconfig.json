{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "hono/jsx",
    "lib": ["DOM"],
    "outDir": "./node_modules/.cache/tsc"
  },
  "extends": "@~/config.typescript/tsx/tsconfig.json",
  "references": [
    {
      "path": "../core"
    },
    {
      "path": "../urls"
    },
    {
      "path": "../../../hono-slotify"
    },
    {
      "path": "../../config/happydom"
    }
  ]
}
