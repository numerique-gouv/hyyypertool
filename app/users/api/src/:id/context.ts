//

import type { App_Context } from "@~/app.middleware/context";
import { urls } from "@~/app.urls";
import type { get_user_by_id_dto } from "@~/users.repository/get_user_by_id";
import type { Env, InferRequestType } from "hono";
import { useRequestContext } from "hono/jsx-renderer";

//

export interface ContextVariablesType extends Env {
  Variables: {
    user: NonNullable<get_user_by_id_dto>;
  };
}
export type ContextType = App_Context & ContextVariablesType;

//

const $get = typeof urls.users[":id"].$get;

type PageInputType = {
  out: InferRequestType<typeof $get>;
};

export const usePageRequestContext = useRequestContext<
  ContextType,
  any,
  PageInputType
>;
