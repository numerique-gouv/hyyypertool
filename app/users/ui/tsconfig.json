{
  "compilerOptions": {
    "jsxImportSource": "hono/jsx",
    "outDir": "./node_modules/.cache/tsc"
  },
  "extends": "@~/config.typescript/api/tsconfig.json",
  "include": ["src"],
  "references": [
    {
      "path": "../../app/ui"
    },
    {
      "path": "../../app/urls"
    },
    {
      "path": "../../moderations/lib"
    },
    {
      "path": "../../organizations/lib"
    },
    {
      "path": "../lib"
    },
    {
      "path": "../../infra/crisp/lib"
    }
  ]
}
