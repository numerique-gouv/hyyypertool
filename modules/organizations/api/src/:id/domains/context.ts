//

import type { App_Context } from "@~/app.middleware/context";
import { urls } from "@~/app.urls";
import type { get_orginization_domains_dto } from "@~/organizations.repository/get_orginization_domains";
import type { Env, InferRequestType } from "hono";
import { useRequestContext } from "hono/jsx-renderer";
import { z } from "zod";

//

const $get = urls.organizations[":id"].domains.$get;

type PageInputType = {
  out: InferRequestType<typeof $get>;
};

//

interface ContextVariablesType extends Env {
  Variables: {
    domains: get_orginization_domains_dto;
  };
}
export type ContextType = App_Context & ContextVariablesType;

//

export const usePageRequestContext = useRequestContext<
  ContextType,
  any,
  PageInputType
>;

export const AddDomainParams_Schema = z.object({ domain: z.string().min(1) });
