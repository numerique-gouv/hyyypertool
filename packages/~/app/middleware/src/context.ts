//

import type { AppVariables_Context } from "@~/app.core/config";
import type { SentryVariables_Context } from "@~/app.sentry";
import type { Csp_Context } from "./csp_headers";
import type { MonComptePro_Pg_Context } from "./moncomptepro_pg";
import type { Session_Context } from "./session";
import type { NonceVariables_Context } from "./set_nonce";
import type { UserInfoVariables_Context } from "./set_userinfo";

export type App_Context = AppVariables_Context &
  Csp_Context &
  MonComptePro_Pg_Context &
  NonceVariables_Context &
  Session_Context &
  UserInfoVariables_Context &
  // CradleVariablesType<object> &
  SentryVariables_Context;
