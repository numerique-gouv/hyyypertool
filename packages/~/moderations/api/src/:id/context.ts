//

import { NotFoundError } from "@~/app.core/error";
import type { App_Context } from "@~/app.middleware/context";
import { urls } from "@~/app.urls";
import { schema, type MonComptePro_PgDatabase } from "@~/moncomptepro.database";
import type { GetFicheOrganizationByIdHandler } from "@~/organizations.lib/usecase";
import { type get_domain_count_dto } from "@~/organizations.repository/get_domain_count";
import { type get_organization_members_count_dto } from "@~/organizations.repository/get_organization_members_count";
import { and, eq } from "drizzle-orm";
import type { Env, InferRequestType } from "hono";
import { useRequestContext } from "hono/jsx-renderer";

//

export interface ModerationContext extends Env {
  Variables: {
    moderation: get_moderation_dto;
  };
}
export interface ContextVariablesType extends Env {
  Variables: {
    domain: string;
    moderation: get_moderation_dto;
    organization_member: get_organization_member_dto;
    organization_fiche: Awaited<ReturnType<GetFicheOrganizationByIdHandler>>;
    query_organization_members_count: Promise<get_organization_members_count_dto>;
    query_domain_count: Promise<get_domain_count_dto>;
  };
}
export type ContextType = App_Context & ContextVariablesType;

//

const $get = typeof urls.moderations[":id"].$get;

type PageInputType = {
  out: InferRequestType<typeof $get>;
};

export const usePageRequestContext = useRequestContext<
  ContextType,
  any,
  PageInputType
>;

//

export async function get_organization_member(
  { pg }: { pg: MonComptePro_PgDatabase },
  { user_id, organization_id }: { user_id: number; organization_id: number },
) {
  return pg.query.users_organizations.findFirst({
    columns: { is_external: true },
    where: and(
      eq(schema.users_organizations.user_id, user_id),
      eq(schema.users_organizations.organization_id, organization_id),
    ),
  });
}
export type get_organization_member_dto = Awaited<
  ReturnType<typeof get_organization_member>
>;

//

export async function get_moderation(
  { pg }: { pg: MonComptePro_PgDatabase },
  { moderation_id }: { moderation_id: number },
) {
  const moderation = await pg.query.moderations.findFirst({
    where: eq(schema.moderations.id, moderation_id),
    with: {
      organization: {
        columns: {
          cached_activite_principale: true,
          cached_adresse: true,
          cached_code_postal: true,
          cached_est_active: true,
          cached_etat_administratif: true,
          cached_libelle_activite_principale: true,
          cached_libelle_categorie_juridique: true,
          cached_libelle_tranche_effectif: true,
          cached_libelle: true,
          cached_nom_complet: true,
          cached_tranche_effectifs: true,
          created_at: true,
          id: true,
          siret: true,
          updated_at: true,
        },
      },
      user: {
        columns: {
          created_at: true,
          email: true,
          family_name: true,
          given_name: true,
          id: true,
          job: true,
          last_sign_in_at: true,
          phone_number: true,
          sign_in_count: true,
        },
      },
    },
  });

  if (!moderation) throw new NotFoundError("Moderation not found.");
  return moderation;
}
export type get_moderation_dto = Awaited<ReturnType<typeof get_moderation>>;
