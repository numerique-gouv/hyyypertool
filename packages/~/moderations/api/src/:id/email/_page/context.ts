//

import type { App_Context } from "@~/app.middleware/context";
import { urls } from "@~/app.urls";
import type { get_crisp_mail_dto } from "@~/crisp.lib";
import type { Crisp_Context } from "@~/crisp.middleware";
import { type get_zammad_mail_dto } from "@~/zammad.lib/get_zammad_mail";
import type { Env, InferRequestType } from "hono";
import { useRequestContext } from "hono/jsx-renderer";
import type { Simplify } from "hono/utils/types";

//

export interface ContextVariablesType extends Env {
  Variables: {
    MAX_ARTICLE_COUNT: number;
    //
    crisp:
      | undefined
      | {
          conversation: get_crisp_mail_dto["conversation"];
          messages: get_crisp_mail_dto["messages"];
          session_id: string;
          show_more: boolean;
          subject: string;
        };
    moderation: Simplify<
      Pick<Moderation, "ticket_id"> & { user: Pick<User, "email"> }
    >;
    zammad:
      | undefined
      | {
          articles: get_zammad_mail_dto;
          show_more: boolean;
          subject: string;
          ticket_id: string;
        };
  };
}
export type ContextType = App_Context & Crisp_Context & ContextVariablesType;

//

const $get = typeof urls.moderations[":id"].email.$get;
type PageInputType = {
  out: InferRequestType<typeof $get>;
};

export const usePageRequestContext = useRequestContext<
  ContextType,
  any,
  PageInputType
>;
