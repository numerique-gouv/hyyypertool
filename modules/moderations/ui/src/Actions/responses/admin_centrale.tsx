//

import { useContext } from "hono/jsx";
import { dedent } from "ts-dedent";
import { context } from "../context";

export const label = "Agent - adresse e-mail départementale —> Admin centrale";

export default function template() {
  const {
    moderation: {
      organization: { cached_libelle: organization_name },
      user: { email },
    },
  } = useContext(context);

  return dedent`
    Bonjour,

    Nous avons bien reçu votre demande de rattachement à l'organisation « ${organization_name} » sur ProConnect (anciennement : AgentConnect, MonComptePro).

    Votre adresse e-mail départementale : « ${email} » ne vous permet pas de rattacher votre compte utilisateur ProConnect à cette administration centrale. 
    Nous vous invitons à créer votre compte utilisateur ProConnect à nouveau, en le rattachant à l'administration déconcentrée dans laquelle vous exercez.

    Bien cordialement,
    L’équipe ProConnect.
  `;
}
