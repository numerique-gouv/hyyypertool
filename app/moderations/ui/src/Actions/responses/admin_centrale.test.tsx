//

import { render_md } from "@~/app.ui/testing";
import { expect, test } from "bun:test";
import { context, type Values } from "../context";
import admin_centrale from "./admin_centrale";

//

test("returns bad email address for admin centrale email", async () => {
  expect(
    await render_md(
      <context.Provider
        value={
          {
            moderation: {
              organization: { cached_libelle: "ðŸ¦„" },
              user: { email: "ðŸ’Œ" },
            },
          } as Values
        }
      >
        <Response />
      </context.Provider>,
    ),
  ).toMatchSnapshot();
});

function Response() {
  return <>{admin_centrale()}</>;
}
