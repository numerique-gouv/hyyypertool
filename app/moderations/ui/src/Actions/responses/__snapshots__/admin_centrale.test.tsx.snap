// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`returns bad email address for admin centrale email 1`] = `
"Bonjour,

Nous avons bien reÃ§u votre demande de rattachement Ã  l&#39;organisation Â« ğŸ¦„ Â» sur ProConnect (anciennementÂ :Â AgentConnect, MonComptePro).

Votre adresse e-mail dÃ©partementaleÂ :Â Â« ğŸ’Œ Â» ne vous permet pas de rattacher votre compte utilisateur ProConnect Ã  cette administration centrale.
Nous vous invitons Ã  crÃ©er votre compte utilisateur ProConnect Ã  nouveau, en le rattachant Ã  l&#39;administration dÃ©concentrÃ©e dans laquelle vous exercez.

Bien cordialement,
Lâ€™Ã©quipe ProConnect.
"
`;
